scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,374 @@
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, IllegalArgumentException {
+
+        FTPClient ftpClient = new FTPClient();
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        //FTP mode is GET
+        if (ftpMode.equals("GET")) {
+            getOut().println("Importing file(s) from " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+            FTPFile[] ftpFile = ftpClient.listFiles(ftpRemoteFilePath);
+            if (ftpFile.length == 0)
+                throw new IllegalArgumentException(ftpRemoteFilePath + " not found. Please, enter a valid path.");
+
+            // If it is a single file:
+            if (ftpFile.length == 1 && ftpRemoteFilePath.contains(ftpFile[0].getName())) {
+                String saveFilePath = Paths.get(ftpLocalFilePath, ftpFile[0].getName()).toString();
+                boolean download = downloadSingleFile(ftpClient, ftpRemoteFilePath, saveFilePath);
+                if (download) {
+                    getOut().println("Succeed to download the file " + ftpRemoteFilePath + " to " + ftpLocalFilePath);
+                } else {
+                    getOut().println("COULD NOT download the file: " + ftpRemoteFilePath);
+                }
+
+                // If the file is a zip, and ftpExtractArchive is set to true
+                if (ftpExtractArchive && ftpRemoteFilePath.endsWith(".zip")) {
+                    ZipFile zipFile = null;
+                    try {
+                        zipFile = new ZipFile(saveFilePath);
+                        // Create an enumeration of the entries in the zip file
+                        Enumeration zipFileEntries = zipFile.entries();
+                        // Process each entry
+                        while (zipFileEntries.hasMoreElements()) {
+                            // grab a zip file entry
+                            ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                            Path path = Paths.get(ftpLocalFilePath, entry.getName());
+                            getOut().println("path: " + path.toString());
+                            if (entry.isDirectory()) {
+                                Files.createDirectories(path);
+                            } else {
+                                Path parentDir = path.getParent();
+                                getOut().println("parentDir of path: " + parentDir.toString());
+                                if (!Files.exists(parentDir)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3725)